coffeecard/CoffeeCard.Library/Services/v2/LeaderboardService.cs

```diff
@@ -23,10 +23,10 @@ public LeaderboardService(CoffeeCardContext context, IDateTimeProvider dateTimeP
 
         public async Task<IEnumerable<LeaderboardEntry>> GetTopLeaderboardEntries(LeaderboardPreset preset, int top)
         {
-            var sortedStatistics = GetSortedStatistics(preset)
+            var sortedStatistics = await GetSortedStatistics(preset)
                 .Take(top)
                 .Include(s => s.User)
-                .AsEnumerable();
+                .ToListAsync();
```

Comment:
I was seeing slightly worse performance on my setup using this instead of enumerable. Presumably because it has to load the entire thing into memory instead. Did you see differently? 

If it is because we want it to be async, I think we should look into the AsyncEnumerable and the nuget package that adds support for linq statements on it.

Reply:
Honestly, it was to asynchronousity.  

After researching a bit, I see a few recommendations to force query evaluation by using `ToListAsync()` at the earliest point where no more filters on the database query are made, see here [[1]](https://anduin.aiursoft.com/post/2020/1/26/tips-to-get-better-performance-for-entity-framework-core) and somewhat related [[2]](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/best-practices?view=aspnetcore-6.0#return-ienumerablet-or-iasyncenumerablet)

Reply:
Did you do any kind of testing regarding the performance? Again, I was seeing better performance on my system with it returning an enumerable, although that is also in a situation where there is only one client making requests, so it might very well be different in production.

Reply:
Per the [2], we shouldn't return an IEnumerable from the controller level, so either here or on the Controller we should apply the ToList()